src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests_DefaultsSource.vb

```diff
@@ -0,0 +1,128 @@
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports System.Collections.Immutable
+Imports Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense.AsyncCompletion
+Imports Microsoft.CodeAnalysis.Host.Mef
+Imports Microsoft.VisualStudio.Language.Intellisense.AsyncCompletion
+Imports Microsoft.VisualStudio.Text.Editor
+
+Namespace Microsoft.CodeAnalysis.Editor.UnitTests.IntelliSense
+    <[UseExportProvider]>
+    Public Class CSharpCompletionCommandHandlerTests_DefaultsSource
+
+        <WpfFact, CombinatorialData>
+        <Trait(Traits.Feature, Traits.Features.Completion)>
+        Public Async Function Test1() As Task
+            Using state = CreateTestState(
+                              <Document>
+using NS1;
+class C
+{
+    void Method()
+    {
+        $$
+    }
+}
+                              </Document>)
+
+                state.SendInvokeCompletionList()
+                Await state.AssertSelectedCompletionItem("MyAB", isHardSelected:=False) ' Not hard-selected since filter text is empty
+            End Using
+        End Function
+
+        <WpfFact, CombinatorialData>
+        <Trait(Traits.Feature, Traits.Features.Completion)>
+        Public Async Function Test2() As Task
+            Using state = CreateTestState(
+                              <Document>
+using NS1;
+class C
+{
+    void Method()
+    {
+        My$$
+    }
+}
+                              </Document>)
+                state.SendInvokeCompletionList()
+                Await state.AssertCompletionItemsContain("MyA", displayTextSuffix:="")
+                Await state.AssertSelectedCompletionItem("MyAB", isHardSelected:=True)
+            End Using
+        End Function
+
+        <WpfTheory, CombinatorialData>
+        <Trait(Traits.Feature, Traits.Features.Completion)>
+        Public Async Function Test3(isAggressive As Boolean) As Task
+            Using state = CreateTestState(
+                              <Document>
+using NS1;
+class C
+{
+    void Method()
+    {
+        MyA$$
+    }
+}
+                              </Document>)
+                If isAggressive Then
+                    state.TextView.Options.SetOptionValue(ItemManager.AggressiveDefaultsMatchingOptionName, True)
+                End If
+
+                state.SendInvokeCompletionList()
+
+                If isAggressive Then
+                    Await state.AssertCompletionItemsContain("MyA", displayTextSuffix:="")
+                    Await state.AssertSelectedCompletionItem("MyAB", isHardSelected:=True)
+                Else
+                    Await state.AssertCompletionItemsContain("MyAB", displayTextSuffix:="")
+                    Await state.AssertSelectedCompletionItem("MyA", isHardSelected:=True)
+                End If
+            End Using
+        End Function
+
+        Private Shared Function CreateTestState(documentElement As XElement) As TestState
```

Comment:
i genuinely find this more confsuing as now all the tests have ambient state that need to be understood.